src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,190 @@
+package duke;
+
+import java.util.Scanner;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+/**
+ * A personal assistant chatbot.
+ */
+public class Duke {
+    private static final String INDENTED_HORIZONTAL_LINE = " ".repeat(4) + "_".repeat(60);
+    /** Platform independent line separator */
+    private static final String LS = System.lineSeparator();
+
+    private static final Scanner SCANNER = new Scanner(System.in);
+
+    private static final String MESSAGE_GREETING = "Hello! I'm Duke" + LS + "What can I do for you?";
+    private static final String MESSAGE_FAREWELL = "Bye. Hope to see you again soon!";
+    private static final String MESSAGE_ERROR = "☹ OOPS!!! %1$s";
+    private static final String MESSAGE_TASK_ADDED = "Got it. I've added this task:" + LS + "  %1$s" + LS
+            + "Now you have %2$s task(s) in the list";
+    private static final String MESSAGE_TASK_LIST = "Here are the tasks in your list:" + LS + "%1$s";
+    private static final String MESSAGE_TASK_MARKED_AS_DONE = "Nice! I've marked this task as done:" + LS + "%1$s";
+
+    private static final String MESSAGE_TODO_DESCRIPTION_EMPTY = "The description of a todo cannot be empty.";
+    private static final String MESSAGE_UNRECOGNISED_COMMAND = "I'm sorry, but I don't know what that means :-(";
+
+    private static final String DEADLINE_PREFIX_BY = "/by";
+    private static final String EVENT_PREFIX_AT = "/at";
+
+    private static final String COMMAND_EXIT = "bye";
+    private static final String COMMAND_ADD_TODO = "todo";
+    private static final String COMMAND_ADD_DEADLINE = "deadline";
+    private static final String COMMAND_ADD_EVENT = "event";
+    private static final String COMMAND_LIST_TASKS = "list";
+    private static final String COMMAND_MARK_TASK_AS_DONE = "done";
+
+    /** Maximum number of tasks possible */
+    private static final int CAPACITY = 100;
+    /** Array of Task objects */
+    private static final Task[] tasks = new Task[CAPACITY];
+    /** Total number of tasks */
+    private static int taskCount = 0;
+
+    /**
+     * Main entry point of Duke.
+     */
+    public static void main(String[] args) {
+        printGreeting();
+        while (true) {
```

Comment:
Suggest to relocate your exit condition to this while loop for better logic clarity.

Reply:
Will implement this in Week 7 as relocating the exit condition requires some refactoring and it makes more sense to do it in conjunction with the stretch goal for `A-MoreOOP`